packages/next/src/server/dev/hot-reloader-webpack.ts

```diff
@@ -1365,6 +1369,7 @@ export default class HotReloaderWebpack implements NextJsHotReloaderInterface {
         changedCSSImportPages.size ||
         reloadAfterInvalidation
       ) {
+        this.resetFetch()
```

Comment:
Can we have some comments about why we only conditionally reset fetch instead of doing somewhere will always get running?

Reply:
because this branch checks for server components changes. I can add a comment tho